src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigProcessing/AnalysisConfigGenerator.cs

```diff
@@ -97,7 +96,10 @@ public static AnalysisConfig GenerateFile(ProcessedArgs localSettings,
             config.SetSettingsFilePath(localSettings.PropertiesFileName);
```

Comment:
I think we should extract a Processor also for the code between the lines 76 and 97. This will allow us to remove the `AddSetting` that is now duplicated and have the full method code written consistently.

Reply:
Great idea!
I went for a bit less than that.
There is a small part of the logic remaining still inside `GenerateFile`, because we would need to pass `additionalSettings` and `buildSettings` to the Processors.

We could do that, but I think it's also okay to only pass the server+local settings and leave the logic inside there.

Reply:
I've modified a bit the structure of the classes to get the state in the constructor. This allowed us to simplify a bit the method signatures since the data doesn't have to be passed around.